kbs/src/resource/backend.rs

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2023 by Alibaba.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::Arc;
+
+use serde::Deserialize;
+
+use super::local_fs;
+use super::{Error, Result};
+
+type RepositoryInstance = Arc<dyn Repository>;
+
+/// Interface of a `Repository`.
+#[async_trait::async_trait]
+pub trait Repository: Send + Sync {
+    /// Read secret resource from repository.
+    async fn read_secret_resource(&self, resource_desc: ResourceDesc) -> Result<Vec<u8>>;
+
+    /// Write secret resource into repository
+    async fn write_secret_resource(&self, resource_desc: ResourceDesc, data: &[u8]) -> Result<()>;
+}
+
+#[derive(Debug, Clone)]
+pub struct ResourceDesc {
+    pub repository_name: String,
+    pub resource_type: String,
+    pub resource_tag: String,
+}
+
+impl ResourceDesc {
+    pub fn is_valid(&self) -> bool {
+        if &self.repository_name == "."
+            || &self.repository_name == ".."
+            || &self.resource_type == "."
+            || &self.resource_type == ".."
```

Comment:
This is for security reason. For example a resource description like `../../secret-file`. Due to the logic of LocalFs it will perform a reading out of the localfs workdir, thus causing unexpected problem

Reply:
oh yeah. that doesn't seem totally rigorous. again maybe we should move this to `try_from` rather than always trying to remember to call the validate code.

Reply:
Fixed.